app/models/host/managed_extensions.rb

```diff
@@ -61,4 +61,9 @@ def delete_discovery_attribute_set
     DiscoveryAttributeSet.destroy_all(:host_id => self.id) if type_changed?
   end
 
+  def update_notifications
+    return if new_record?
+    return unless type_changed?
+    ForemanDiscovery::UINotifications::DestroyHost.deliver!(self)
```

Comment:
Actually, scratch that, I did create a exception handling in core, so this will only warn vs raise.